compiler/ast/parser.nim

```diff
@@ -1358,22 +1324,30 @@ proc postExprBlocks(p: var Parser, x: PNode): PNode =
     p.getTok
     p.skipComment(result)
     if p.tok.tokType notin {tkOf, tkElif, tkElse, tkExcept}:
-      var stmtList = newTreeI(nkStmtList, p.lineInfo, parseStmt(p))
+      var stmtList = p.newNode(nkStmtList, @[parseStmt(p)])
       # to keep backwards compatibility (see tests/vm/tstringnil)
       if stmtList[0].kind == nkStmtList: stmtList = stmtList[0]
 
-      stmtList.flags.incl nfBlockArg
+      stmtList.isBlockArg = true
+      # stmtList.flags.incl nfBlockArg # XXXX
       result.add if openingParams.kind == nkEmpty and openingPragmas.kind == nkEmpty:
                    stmtList
                  else:
-                   newProcNode(nkDo, stmtList.info, body = stmtList,
+                   newProcNode(
+                     nkDo,
+                     stmtList.info,
+                     body = stmtList,
                      params = if openingParams.kind != nkEmpty: openingParams
-                              else: newTreeI(nkFormalParams, p.lineInfo, p.emptyNode),
-                     name = p.emptyNode, pattern = p.emptyNode, genericParams = p.emptyNode,
-                     pragmas = openingPragmas, exceptions = p.emptyNode)
+                              else: p.newNode(nkFormalParams, @[p.emptyNode]),
```

Comment:
I think this is missing line info

Reply:
line information is added automatically from the parser in

```nim
proc newNode(
    p: Parser,
    kind: TNodeKind,
    subnodes: sink seq[ParsedNode] = @[],
    info: TLineInfo = p.lineInfo
  ): ParsedNode =
  ## Create a new non-leaf parsed node using line information from the
  ## parser p
  return newParsedNode(kind, info, subnodes)
```

Reply:
`else: newTreeI(nkFormalParams, p.lineInfo, p.emptyNode),` -> `.newNode(nkFormalParams, @[p.emptyNode]),`